API_DOCUMENTATION.md

```diff
@@ -0,0 +1,217 @@
+# Noctua Web App API Documentation
+
+This document provides comprehensive documentation for the Noctua Web App API endpoints.
+
+## Authentication
+
+All API endpoints require authentication using NextAuth.js. Users must be signed in with a valid session token to access these endpoints.
+
+### Authentication Endpoint
+
+#### `POST /api/auth/[...nextauth]`
+
+Handles authentication using NextAuth.js with Google provider.
+
+- **Access Control**: Only users with allowed email addresses can sign in.
+- **Response**: Returns authentication tokens and session information.
+
+## Conversations API
+
+### Create a New Conversation
+
+#### `POST /api/conversations`
+
+Creates a new conversation for the authenticated user.
+
+- **Authentication**: Requires a valid session token.
+- **Request Body**: Empty object `{}`.
+- **Response**:
+  - **Status Code**: 201 (Created)
+  - **Body**:
+    ```json
+    {
+      "conversationId": "string"
+    }
+    ```
+- **Error Responses**:
+  - 401: Unauthorized (Invalid session token)
+  - 500: Failed to create conversation
+
+### Get User Conversations
+
+#### `GET /api/conversations`
+
+Retrieves all conversations for the authenticated user.
+
+- **Authentication**: Requires a valid session token.
+- **Response**:
+  - **Status Code**: 200 (OK)
+  - **Body**:
+    ```json
+    {
+      "conversations": [
+        {
+          "id": "string",
+          "messages": [],
+          "created_at": "timestamp",
+          "updated_at": "timestamp",
+          "title": "string",
+          "user_id": "string"
+        }
+      ]
+    }
+    ```
+- **Error Responses**:
+  - 401: Unauthorized (Invalid session token)
+  - 404: User not found
+  - 500: Failed to get conversations
+
+### Delete a Conversation
+
+#### `DELETE /api/conversations/[conversationId]`
+
+Deletes a specific conversation.
+
+- **Authentication**: Requires a valid session token.
+- **URL Parameters**:
+  - `conversationId`: ID of the conversation to delete
+- **Response**:
+  - **Status Code**: 200 (OK)
+  - **Body**:
+    ```json
+    {
+      "message": "Conversation deleted successfully"
+    }
+    ```
+- **Error Responses**:
+  - 401: Unauthorized (Invalid session token)
+  - 403: You do not have access to this conversation
+  - 404: User not found or Conversation not found
+  - 500: Failed to delete conversation
+
+## Messages API
+
+### Send a Message
+
+#### `POST /api/conversations/[conversationId]/messages`
+
+Sends a message to a specific conversation and gets a response from the LLM Engine.
+
+- **Authentication**: Requires a valid session token.
+- **URL Parameters**:
+  - `conversationId`: ID of the conversation
+- **Request Body**:
+  ```json
+  {
+    "message": "string"
+  }
+  ```
+- **Response**:
+  - **Status Code**: 201 (Created)
+  - **Body**:
+    ```json
```

Comment:
I think if you change the code blocks from json to javascript then you can have comments, bools etc. At the moment they are being highlighted as json errors. Worth a try.